content/en/blog/2025/otel-collector-k8s-discovery/index.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Kubernetes annotation based discovery for OpenTelemetry Collector
+linkTitle: Kubernetes annotation discovery
+date: 2025-01-20
+author: >
+  [Dmitrii Anoshin](https://github.com/dmitryax) (Cisco/Splunk), [Christos
+  Markou](https://github.com/ChrsMark) (Elastic)
+cSpell:ignore: Dmitrii Anoshin Markou
+---
+
+In the world of containers and Kubernetes, observability is crucial. Users need
+to know the status of their workloads at any given time. In other words, they
+need observability into moving objects.
+
+This is where the [OpenTelemetry Collector](/docs/collector) and its
+[receiver creator](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.117.0/receiver/receivercreator)
+component come in handy. Users can set up fairly complex monitoring scenarios
+with a self-service approach, following the principle of least privilege at the
+cluster level.
+
+The self-service approach is great, but how much self-service can it actually
+be? In this blog post, we will explore a newly added feature of the Collector
+that makes dynamic workload discovery even easier, providing a seamless
+experience for both administrators and users.
+
+## What is Autodiscovery in Observability?
+
+Applications running on containers and pods become moving targets for the
+monitoring system. With autodiscovery, monitoring agents like the Collector can
+track changes at the container and pod levels and dynamically adjust the
+monitoring configuration.
+
+Today, the Collector—and specifically the receiver creator—can provide such an
+experience. Using the receiver creator, observability users can define
+configuration "templates" that rely on environment conditions. For example, as
+an observability engineer, I can configure my Collector to enable the
+[Redis receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.117.0/receiver/redisreceiver)
+when a Redis pod is deployed on the cluster. The following configuration can
```

Comment:
maybe I am very pedantic here, but since Redis is no longer open source, wouldn't it be better to pick something wie an OSI-approved open source license for the example?

Reply:
I can use NGINX which is our example further down as well.

Reply:
yes, that would be great!

Reply:
Fixed.